substrate/utils/wasm-builder/src/lib.rs

```diff
@@ -327,6 +334,37 @@ impl CargoCommand {
 		// Check if major and minor are greater or equal than 1.68 or this is a nightly.
 		version.major > 1 || (version.major == 1 && version.minor >= 68) || version.is_nightly
 	}
+
+	/// Returns whether this version of the toolchain supports the `wasm32v1-none` target.
+	fn supports_wasm32v1_none_target(&self) -> bool {
+		self.version.map_or(false, |version| {
+			// Check if major and minor are greater or equal than 1.84.
+			version.major > 1 || (version.major == 1 && version.minor >= 84)
+		})
+	}
+
+	fn is_target_installed(toolchain: &str, target: &str) -> bool {
+		Command::new("rustup")
```

Comment:
https://users.rust-lang.org/t/determining-installed-target-list-when-rustup-is-not-in-use/100594/2

Please use this combined with `DummyCrate::().syroot` and then check if the folder exists. This way it doesn't require `rustup`.

Reply:
done